### PR TITLE
Cleanup and handleSummaryResult() tests

### PR DESCRIPTION
**Pull request by na--**
There were a few other things I wondered if I should fix in the current Go summary code, but at this point in time I'll probably just fix them in the upcoming JS replacement... For example, this code: https://github.com/loadimpact/k6/blob/c00a85bfea21d748f0f9d11fdbc326087f34c411/ui/summary.go#L91 

Should be
```go
if !inEscSeq && !inLongEscSeq {
	n++
}
```
i.e. that `StrWidth()` function is completely broken... :man_facepalming: But this turns not to matter one bit, since almost all of the errors are by the same margin and mostly cancel out... :rofl: Not to mention that the Unicode normalization it tries to do might actually be worse than not doing anything at all: https://github.com/loadimpact/k6/blob/c00a85bfea21d748f0f9d11fdbc326087f34c411/ui/summary.go#L64-L65

See [this MDN article](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/normalize) for a good explanation of the various normalization modes, and why `NFC` or `NFKC` (or even doing nothing) might be better for our use case than `NFKD`. On the other hand, not even they are sufficient, so maybe it doesn't really matter and everything in life is pointless: https://hsivonen.fi/string-length/ :sob: :sweat_smile: :rofl: 

Anyway, I think this is the final summary-related PR before v0.30.0.

### CODE REVIEW

**Review by MStoykov — APPROVED** [review]
LGTM!

 I opted to not read the linked stuff ... for now :P

**Comment by na--** [comment]
You are very wise... :sweat_smile: ugh, I'll have to remember how to fix the windows fs tests now...

File: js/summary.go [hunk 1]
```diff
@@ -36,10 +36,12 @@ import (
...
 const summaryWrapperLambdaCode = `
 (function() {
-	var forEach = function(obj, callback) {
+	var forEach = function (obj, callback) {
 		for (var key in obj) {
 			if (obj.hasOwnProperty(key)) {
-				callback(key, obj[key]);
+				if (callback(key, obj[key])) {
+					break;
+				}
```
Comment (imiric) [inline js/summary.go:44]: Why was this added? I don't see any of the callbacks returning `true`.

Would it make more sense to `catch` an error to trigger this `break` instead?

Comment (na--) [reply js/summary.go:44]: It was added for the text summary generation code. It currently lives in https://gist.github.com/na--/eaf6540fae765499cbfddce1a052583f, but I'll clean it up and push it to jslib.k6.io soon, and it will be bundled in the k6 v0.31.0 cycle, so the `forEach` changes will soon be used. And anyway, returning a result seems cleaner than throwing an error - for one thing, I wouldn't want to catch other actual errors by mistake.

**Review by imiric — APPROVED** [review]

**Review by imiric — APPROVED** [review]
